UnityProject/Assets/Scripts/Items/Others/LightPreparable.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Threading.Tasks;
+using NaughtyAttributes;
+using Systems.Explosions;
+using UI.Action;
+using UnityEngine;
+
+namespace Items.Others
+{
+	[RequireComponent(typeof(ItemLightControl))]
+	public class LightPreparable : HandPreparable
+	{
+		private ItemLightControl lightControl;
+		private SpriteDataSO offSprite;
+		[SerializeField, Tooltip("GlowSprite's handlers cannot be automatically fetched from Awake()")]
+		private SpriteHandler glowHandler;
+		[SerializeField, Tooltip("Create a spark when activating this light source?")]
+		private bool sparkOnOpen = false;
+		[SerializeField]
+		private bool runsOutOverTime = false;
+		[SerializeField, ShowIf(nameof(runsOutOverTime))]
+		private float runsOutOverSeconds = 356;
+		[SerializeField] private ItemActionButton actionButton;
+
+
+		private void Awake()
+		{
+			lightControl = GetComponent<ItemLightControl>();
+			if (runsOutOverTime)
+			{
+				destroyThisComponentOnOpen = false;
+				offSprite = spriteHandler.GetCurrentSpriteSO();
+			}
+			if(glowHandler != null) glowHandler.SetActive(false);
+			if(actionButton == null) return;
+			actionButton.ServerActionClicked += Open;
+		}
+
+		public override void Open()
+		{
+			lightControl.Toggle(true);
+			if(sparkOnOpen) SparkUtil.TrySpark(gameObject);
+			if(runsOutOverTime) LightsOutAfterTime(lightControl, spriteHandler);
+			if(glowHandler != null) glowHandler.SetActive(true);
+			actionButton.ServerActionClicked -= Open;
+			base.Open();
+		}
+
+		private async void LightsOutAfterTime(ItemLightControl lightController, SpriteHandler handler)
+		{
+			await Task.Delay((int)runsOutOverSeconds * 1000);
+			lightController.Toggle(false);
+			handler.SetSpriteSO(offSprite);
```

Comment:
This might complain since it's technically another thread after the await

Reply:
this will complain with mirror*

Reply:
Wouldn't giving a direct reference to the handlers solve this issue?

Reply:
eh, don't want to risk another surprise thread error. Replaced it with `IEnumerator`.